pyqt-apps/siriushla/bl_ap_imgproc/main.py

```diff
@@ -243,6 +252,109 @@ def _setupTab(self, content, use_scroll=False):
             return sc_area
         return cont_wid
 
+    def toggle_beamline_btns(self, value):
+        if value == 0:
+            state = True
+        else:
+            state = False
+
+        self.end_processing_cmd()
+        self.open_beamline_btn.setEnabled(state)
+
+    def end_processing_cmd(self):
+        self.enable_gamma_btn.setEnabled(True)
+        self.disable_gamma_btn.setEnabled(True)
+        self.open_beamline_btn.setEnabled(True)
+        self.loading.setVisible(False)
+
+    def start_processing_cmd(self):
+        self.enable_gamma_btn.setEnabled(False)
+        self.disable_gamma_btn.setEnabled(False)
+        self.open_beamline_btn.setEnabled(False)
+        self.loading.setVisible(True)
+
+    def intlk_cmd(self, cmd):
+        self.start_processing_cmd()
+        if cmd == "disable_gamma":
+            self.blpps.gamma_disable()
+        elif cmd == "enable_gamma":
+            self.blpps.gamma_enable()
+        elif cmd == "open_beamline":
+            self.blpps.beamline_open()
+
+    def gamma_control(self):
+        wid = QGroupBox()
+        lay = QHBoxLayout()
+        wid.setLayout(lay)
+        wid.setTitle("Gamma")
+        wid.setMaximumHeight(200)
+
+        self.enable_gamma_btn = QPushButton("Enable")
+        self.enable_gamma_btn.clicked.connect(
+            lambda: self.intlk_cmd("enable_gamma"))
+        lay.addWidget(self.enable_gamma_btn)
+
+        self.disable_gamma_btn = QPushButton("Disable")
+        self.disable_gamma_btn.clicked.connect(
+            lambda: self.intlk_cmd("disable_gamma"))
+        lay.addWidget(self.disable_gamma_btn)
+
+        pvname = INTLK_PVS["gamma"]
+        widget = SiriusLedState(init_channel=pvname)
+        self.gamma_enabled_conn = SiriusConnectionSignal(pvname)
+        self.gamma_enabled_conn.new_value_signal[int].connect(
+            self.toggle_beamline_btns)
+        lay.addWidget(widget)
+
+        return wid
+
+    def update_bl_open_led(self):
+        if self.pydm_led != None:
+            status_bl = self.blpps.beamline_opened
+            old_val = self.pydm_led.value
+            self.pydm_led.value_changed(status_bl)
+            if old_val != status_bl:
+                self.end_processing_cmd()
+
+    def enable_beamline(self):
```

Comment:
maybe the name of this method could be more descriptive: _setup_enable_beamline_widgets